src/ruby/bin/interop/interop_client.rb

```diff
@@ -228,6 +228,26 @@ def compute_engine_creds
     p 'OK: compute_engine_creds'
   end
 
+  def oauth2_auth_token
+    auth_creds = Google::Auth.get_application_default(@args.oauth_scope)
+    kw = auth_creds.updater_proc.call({})
+    resp = perform_large_unary(fill_username: true,
+                               fill_oauth_scope: true,
+                               **kw)
+    json_key = File.read(ENV[AUTH_ENV])
+    wanted_email = MultiJson.load(json_key)['client_email']
+    assert_equal(wanted_email, resp.username,
+                "#{__callee__}: incorrect username")
+    assert(@args.oauth_scope.include?(resp.oauth_scope),
+           "#{__callee__}: incorrect oauth_scope")
+    p "OK: #{__callee__}"
+  end
+
+  # Due to the functional interface of the Credentials implementation used by
+  # google-auth-library ruby, per_rpc_creds and oauth2_auth_token are
+  # fundamentally the same test.
```

Comment:
In my implementation, `oauth2_auth_token` uses a metadata updater function attached to the stub, and `per_rpc_creds` directly passes the metadata to the call. Why not do that?